### ISSUE TITLE
FONV LOD BSSegment 

### ISSUE THREAD

**Issue by neomonkeus** [niftools/nifxml#38]
Migrated from sf.net - @ttl269, @deedes

This seems to be correct decode for BSSegmentedTriangle, at least as far as Fallout New Vegas LOD data is conserned. 
It relates to the number of triangles in the triangle data per 'segment'
The segments seem to be limited to a max of 16, or 4x4 cells(?)
Confirmed by generating a new worldspace, placing 1 object in the worldspace and generating object lod for that worldspace
then repeating with more objects and noting the triangle count of the objects vs the resulting object lod files and BSSegmentedTriangle section.

``` xml
<compound name="BSSegmentedTriangle">
Bethesda-specific node.
<add name="Unknown Byte 1" type="byte" >Unknown</add>
<add name="triangle*3 offset" type="short" >triangle*3 offset</add>
<add name="Unknown Byte 2" type="byte" >Unknown</add>
<add name="Unknown Byte 3" type="byte" >Unknown</add>
<add name="triangle count" type="short" >triangle count</add>
<add name="Unknown Byte 4" type="byte" >Unknown</add>
<add name="Unknown Byte 5" type="byte" >Unknown</add>
</compound>
```

Sorry for the late reply. From what I can infer from the history, our current nif.xml calls it BSSegment:

``` xml
<enum name="BSSegmentFlags" storage="uint">
An unsigned 32-bit integer, describing what's inside the segment.
<option value="9" name="BSSEG_WATER">Contains water.</option>
</enum>
<compound name="BSSegment">
Bethesda-specific node.
<add name="Internal index" type="int" >Index multiplied by 1536 (0x0600)</add>
<add name="Flags" type="BSSegmentFlags" default="0x00000000">Geometry present in the segment</add>
<add name="Unknown Byte 1" type="byte" >Unknown</add>
</compound>
<niobject name="BSSegmentedTriShape" inherit="NiTriShape">
Bethesda-specific node.
<add name="Num Segments" type="int" >Number of segments in the square grid</add>
<add name="Segment" type="BSSegment" arr1="Num Segments">Configuration of each segment</add>
</niobject>
```

What you're suggesting appears to be something vastly different. Do you think you can produce a patch against our current xml? Note that it is in process of a severe cleanup operation; you'll find the latest one here:
https://github.com/amorilia/nifxml/tree/feature/history-cleanup

Sorry but I have absolutely no clue how to edit XML.
I provide this information on an AS IS basis and don't mod FONV anymore.
No idea how you 'rename' things and still manage for the nifskope to link to the correct nif chunk.
I do however know thats what BSSegmentFlags does in FONV has nothing to do with flags or water or multiplying by 0600. It works out to the exact tris count of LOD objects per 'cell' and offset in triangles.

errr not BSSegmentFlags but BSSegmentedTriangle
Used to be between

`````` xml
<niobject name="BSMultiBoundSphere" inherit="BSMultiBoundData">```
and 
```xml
<niobject name="BSSegmentedTriShape" inherit="NiTriShape">
``````

If that means anything to you.
And yes, I understand its vastly diffrent but I generated a brand new worldspace and played with LOD objects for a few hours to come up with this decode. No idea exactly what games it applys to, but Im sure it applys to FONV.

Ok, I'll revisit this when the cleanup is done. It should be pretty easy to integrate your updates. I'm sure our current xml is vastly wrong, as the current decoding of that block just seems weird.

Yea, Maybe the current decoding makes sense in some other game, but it does not make any sense in the FONV lod files that I found this block in.

**Comment by ttl269** [comment]
@neomonkeus @deedes
Someone's research described on top is correct. Present structure of `BSSegment` in nif.xml is completely wrong. I have changed it according to informations here and additional testing (#42). It seems that both "triangle offset" and "triangle count" can be stored as `int`, so then there are no additional *unknown bytes* needed. Only fist byte is *unknown* and after opening a lot of Fallout 3 and Skyrim files it seems that it is always set to 0.

**Comment by jonwd7** [comment]
This was addressed with #56 (which closed #42).   Indeed the stuff in the XML was completely wrong, such as the index*1536 and the position/size of the flags.  These observations were a coincidence because they were looking at the wrong groups of bits.

Since this has been addressed by #42 and #56 I am closing.
